File: src/frontend/src/api/hr/payroll.ts
/**
 * M3 인사/급여 — 급여 API 호출 함수
 */
import api from '../client';

export interface PayrollDetail {
  id: string;
  employee_id: string;
  employee_no: string | null;
  employee_name: string | null;
  department_name: string | null;
  base_salary: number;
  overtime_pay: number;
  bonus: number;
  meal_allowance: number;
  car_allowance: number;
  research_allowance: number;
  childcare_allowance: number;
  other_allowance: number;
  gross_salary: number;
  taxable_salary: number;
  income_tax: number;
  local_tax: number;
  national_pension: number;
  health_insurance: number;
  long_term_care: number;
  employment_insurance: number;
  total_deduction: number;
  net_salary: number;
  work_days: number;
  leave_days: number;
  absent_days: number;
  leave_deduction: number;
  ai_optimized: boolean;
}

export interface PayrollHeader {
  id: string;
  payroll_year: number;
  payroll_month: number;
  status: string;
  total_employees: number;
  total_gross: number;
  total_deduction: number;
  total_net: number;
  payment_date: string | null;
  journal_id: string | null;
  approved_by: string | null;
  approved_at: string | null;
  created_at: string;
  details?: PayrollDetail[];
}

export async function fetchPayrolls(year?: number) {
  const res = await api.get('/hr/payroll', { params: year ? { year } : {} });
  return res.data.data as PayrollHeader[];
}

export async function fetchPayroll(year: number, month: number) {
  const res = await api.get(`/hr/payroll/${year}/${month}`);
  return res.data.data as PayrollHeader | null;
}

export async function calculatePayroll(year: number, month: number) {
  const res = await api.post('/hr/payroll/calculate', {
    payroll_year: year,
    payroll_month: month,
  });
  return res.data;
}

export async function approvePayroll(year: number, month: number, paymentDate?: string) {
  const res = await api.post(`/hr/payroll/${year}/${month}/approve`, {
    payment_date: paymentDate || null,
  });
  return res.data;
}

File: src/backend/modules/m3_hr/schemas/payroll.py
"""
M3 인사/급여 — 급여 스키마
"""
from datetime import date, datetime
from typing import Optional, List
from pydantic import BaseModel, Field


class PayrollCalculateRequest(BaseModel):
    """급여 계산 요청"""
    payroll_year: int = Field(..., description="귀속 연도")
    payroll_month: int = Field(..., ge=1, le=12, description="귀속 월")


class PayrollDetailResponse(BaseModel):
    """급여 상세 내역 (1인)"""
    id: str
    employee_id: str
    employee_no: Optional[str] = None
    employee_name: Optional[str] = None
    department_name: Optional[str] = None

    # 지급 항목
    base_salary: float
    overtime_pay: float
    bonus: float
    meal_allowance: float
    car_allowance: float
    research_allowance: float
    childcare_allowance: float
    other_allowance: float
    gross_salary: float
    taxable_salary: float

    # 공제 항목
    income_tax: float
    local_tax: float
    national_pension: float
    health_insurance: float
    long_term_care: float
    employment_insurance: float
    total_deduction: float
    net_salary: float

    # 근태
    work_days: int
    leave_days: float
    absent_days: float
    leave_deduction: float

    ai_optimized: bool

    model_config = {"from_attributes": True}


class PayrollHeaderResponse(BaseModel):
    """급여 대장 헤더 응답"""
    id: str
    payroll_year: int
    payroll_month: int
    status: str
    total_employees: int
    total_gross: float
    total_deduction: float
    total_net: float
    payment_date: Optional[date] = None
    journal_id: Optional[str] = None
    approved_by: Optional[str] = None
    approved_at: Optional[datetime] = None
    created_at: Optional[datetime] = None
    details: Optional[List[PayrollDetailResponse]] = None

    model_config = {"from_attributes": True}


class PayrollApproveRequest(BaseModel):
    """급여 승인 요청"""
    payment_date: Optional[date] = Field(None, description="지급일")
